src/main/kotlin/no/nav/syfo/identhendelse/IdenthendelseService.kt

```diff
@@ -0,0 +1,69 @@
+package no.nav.syfo.identhendelse
+
+import kotlinx.coroutines.runBlocking
+import no.nav.syfo.application.database.DatabaseInterface
+import no.nav.syfo.behandlendeenhet.database.getPersonByIdent
+import no.nav.syfo.client.pdl.PdlClient
+import no.nav.syfo.domain.PersonIdentNumber
+import no.nav.syfo.identhendelse.database.deletePerson
+import no.nav.syfo.identhendelse.database.updatePerson
+import no.nav.syfo.identhendelse.kafka.COUNT_KAFKA_CONSUMER_PDL_AKTOR_UPDATES
+import no.nav.syfo.identhendelse.kafka.KafkaIdenthendelseDTO
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class IdenthendelseService(
+    private val database: DatabaseInterface,
+    private val pdlClient: PdlClient,
+) {
+
+    private val log: Logger = LoggerFactory.getLogger(IdenthendelseService::class.java)
+
+    fun handleIdenthendelse(identhendelse: KafkaIdenthendelseDTO) {
+        if (identhendelse.folkeregisterIdenter.size > 1) {
+            val activeIdent = identhendelse.getActivePersonident()
+            if (activeIdent != null) {
+                val inactiveIdenter = identhendelse.getInactivePersonidenter()
+                val oldPersonIdentList = inactiveIdenter.mapNotNull { personident ->
+                    val person = database.getPersonByIdent(personident)
+                    if (person != null) PersonIdentNumber(person.personident) else null
+                }
+
+                if (oldPersonIdentList.isNotEmpty()) {
+                    checkThatPdlIsUpdated(activeIdent)
+                    val numberOfUpdatedIdenter = updatePersonOrDeleteOldVersions(activeIdent, oldPersonIdentList)
+                    log.info("Identhendelse: Updated $numberOfUpdatedIdenter rows based on Identhendelse from PDL")
+                    COUNT_KAFKA_CONSUMER_PDL_AKTOR_UPDATES.increment(numberOfUpdatedIdenter.toDouble())
+                }
+            } else {
+                log.warn("Mangler gyldig ident fra PDL")
+            }
+        }
+    }
+
+    private fun updatePersonOrDeleteOldVersions(
+        activeIdent: PersonIdentNumber,
+        oldPersonIdentList: List<PersonIdentNumber>
+    ): Int {
+        var updatedRows = 0
+        val personActiveIdent = database.getPersonByIdent(activeIdent)
+        if (personActiveIdent != null) {
+            var deletedRows = 0
+            oldPersonIdentList.forEach { deletedRows += database.deletePerson(it) }
+            log.info("Identhendelse: Deleted $deletedRows entries with an inactive personident from database.")
+        } else {
+            updatedRows += database.updatePerson(activeIdent, oldPersonIdentList.first())
+        }
+        return updatedRows
+    }
+
+    // Erfaringer fra andre team tilsier at vi burde dobbeltsjekke at ting har blitt oppdatert i PDL før vi gjør endringer
+    private fun checkThatPdlIsUpdated(nyIdent: PersonIdentNumber) {
```

Comment:
hva skjer hvis det ikke er oppdatert i PDL? Bare hopper vi over den endringa?

Reply:
Den er enten oppdatert, eller så kaster vi feil, og den går inn i retry-loop 👍🏼 
Jeg gjorde også en endring i siste commit, fordi logikken blir litt annerledes når man leser topicen fra starten 🤔

Reply:
Da kastes en feil og vi prøver på nytt 1 minutt senere.
Vi har også den casen der vi leser topicen fra starten av, og vi får inn en identhendelse på en ident som kommer til å endre seg i fremtiden (en senere record). Da vil `activeIdent` fra record egentlig være utdatert i "real-time". Derfor ble det en litt annen logikk her nå for å ta høyde for den casen.